cmd/habitat/community/executor.go

```diff
@@ -0,0 +1,54 @@
+package community
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/eagraf/habitat/cmd/habitat/community/state"
+	"github.com/eagraf/habitat/cmd/habitat/procs"
+	"github.com/eagraf/habitat/pkg/compass"
+	"github.com/rs/zerolog/log"
+)
+
+type CommunityExecutor struct {
+	processManager *procs.Manager
+}
+
+func NewCommunityExecutor(p *procs.Manager) *CommunityExecutor {
+	return &CommunityExecutor{
+		processManager: p,
+	}
+}
+
+func (e *CommunityExecutor) Execute(update *state.StateUpdate) {
+	log.Info().Msgf("executing %s state transition", update.TransitionType)
+	newState, err := update.State()
+	if err != nil {
+		log.Error().Msgf("error getting new state: %s", err)
+	}
+	switch update.TransitionType {
+	case state.TransitionTypeInitializeIPFSSwarm:
```

Comment:
nice. (kinda answers my q from before). i think the naming of this is kind of confusi,g but this handles updates to community state machine right? also are we thinking that evry type of transition gets listed out here?

Reply:
yeah eventually there will be a lot more types of transitions. Also lmk whats confusing about the name so I can fix